version: "3.7"
services:

  nginx:
    image: nginx:latest
    container_name: nginx
    ports:
      - "8080:80"
    volumes:
      - ./nginx/nginx.conf:/etc/nginx/nginx.conf
    depends_on:
      - node-fc
    networks:
      - desafio-network

  # node:
  #   image: node
  #   networks:
  #     - desafio-network
  #   ports:
  #     - "3000:3000"
  #   volumes:
  #     - ./node:/usr/src/app
  #   tty: true
  #   command: node /usr/src/app/index.js
  #   command: node /usr/src/app/index.js


  mysql:
    container_name: mysql
    image: mysql:latest
    command: --innodb-use-native-aio=0 #?
    restart: always # ?
    tty: true # ?
    volumes:
      - ./mysql:/var/lib/mysql

    environment:
        - MYSQL_DATABASE=desafio
        - MYSQL_ROOT_PASSWORD=1234

    networks:
      - desafio-network
    

  node-fc:
    container_name: node-fc
    build: ./node
    ports:
      - "3000:3000"
    networks:
      - desafio-network
    depends_on:
      mysql:
        condition: service_healthy
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:3000/"]
      interval: 30s
      timeout: 10s
      retries: 3
      start_period: 5s
networks:
  desafio-network:
    driver: bridge


 
